x-pack/test/detection_engine_api_integration/security_and_spaces/group10/perform_bulk_action.ts

```diff
@@ -2997,5 +3003,402 @@ export default ({ getService }: FtrProviderContext): void => {
+          },
+        } = await getRuleSOById(es, ruleWithLegacyInvestigationField.id);
+
+        expect(ruleSOOriginalLegacy?.alert?.params?.investigationFields).to.eql([
+          'client.address',
+          'agent.name',
+        ]);
+
+        const {
+          hits: {
+            hits: [{ _source: ruleSOOriginalLegacyEmptyArray }],
+          },
+        } = await getRuleSOById(es, ruleWithLegacyInvestigationFieldEmptyArray.id);
+        expect(ruleSOOriginalLegacyEmptyArray?.alert?.params?.investigationFields).to.eql([]);
+
+        const {
+          hits: {
+            hits: [{ _source: ruleSOOriginalNoLegacy }],
+          },
+        } = await getRuleSOById(es, ruleWithIntendedType.id);
+        expect(ruleSOOriginalNoLegacy?.alert?.params?.investigationFields).to.eql({
+          field_names: ['host.name'],
+        });
+      });
+
+      it('should edit rules with legacy investigation fields', async () => {
+        const { body } = await postBulkAction().send({
+          query: '',
+          action: BulkActionType.edit,
+          [BulkActionType.edit]: [
+            {
+              type: BulkActionEditType.set_tags,
+              value: ['reset-tag'],
+            },
+          ],
+        });
+        expect(body.attributes.summary).to.eql({
+          failed: 0,
+          skipped: 0,
+          succeeded: 3,
+          total: 3,
+        });
+
+        // Check that the updated rule is returned with the response
+        // and field transformed on response
+        const ruleWithLegacyField = body.attributes.results.updated.find(
+          (returnedRule: RuleResponse) =>
+            returnedRule.rule_id === ruleWithLegacyInvestigationField.params.ruleId
+        );
+        expect(ruleWithLegacyField.investigation_fields).to.eql({
+          field_names: ['client.address', 'agent.name'],
+        });
+        expect(ruleWithLegacyField.tags).to.eql(['reset-tag']);
+
+        const ruleWithEmptyArray = body.attributes.results.updated.find(
+          (returnedRule: RuleResponse) =>
+            returnedRule.rule_id === ruleWithLegacyInvestigationFieldEmptyArray.params.ruleId
+        );
+        expect(ruleWithEmptyArray.investigation_fields).to.eql(undefined);
+        expect(ruleWithEmptyArray.tags).to.eql(['reset-tag']);
+
+        const ruleWithIntendedType = body.attributes.results.updated.find(
+          (returnedRule: RuleResponse) => returnedRule.rule_id === 'rule-with-investigation-field'
+        );
+        expect(ruleWithIntendedType.investigation_fields).to.eql({ field_names: ['host.name'] });
+        expect(ruleWithIntendedType.tags).to.eql(['reset-tag']);
+
+        /**
+         * Confirm type on SO so that it's clear in the tests whether it's expected that
+         * the SO itself is migrated to the inteded object type, or if the transformation is
+         * happening just on the response. In this case, change should not include a migration on SO.
+         */
```

Comment:
I'm updating the description of the ticket with a table that specifies the expected migration/transform behavior for all the routes.

It's not migrating right now, but maybe it makes sense to since PUT/PATCH rule endpoints do. @banderror any thoughts here?

Reply:
IMO it would be simpler to avoid having routes migrate the internal data structure in most cases. I don't think there's a big advantage to a migrated SO vs a legacy SO since I wouldn't expect the new structure to be more performant and we need to handle both structures in the server side code anyway since we can't guarantee that all SOs will ever be migrated.

With that line of thinking, `import` could be the only special case that does migrate the structure on the way in to allow importing rules from 8.10. Every other route would convert the data before returning it, but would not have to migrate the stored data format.

Reply:
I'm fine with @marshallmain's suggestion because it simplifies the code a lot.